File: BOJ/1541 잃어버린 괄호.py
n = input().split('-') #-를 기준으로 리스트로 나눔 / 2 + 3 + 5 - 1 + 7 - 8

#n[0] = 2 + 3 + 5
ans = sum(map(int, n[0].split('+'))) #-이전의 값은 +를 기준으로 나눠서 더함

#- 이후의 다 더할 값들 순회 후 빼기
for i in n[1:]:
    ans -= sum(map(int, i.split('+')))

print(ans)

File: BOJ/1541 잃어버린 괄호.java
import java.util.Scanner;

public class 잃어버린괄호 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        String[] parts = input.split("-");
        int result = Integer.parseInt(parts[0]);

        for (int i = 1; i < parts.length; i++){
            result -= sum(parts[i]);
        }

        System.out.print(result);
    }

    public static int sum(String str){
        //String을 담을 수 있는 배열
        String[] numbers = str.split("\\+");
        int sum = 0;
        for (String number : numbers){
            sum += Integer.parseInt(number);
        }

        return sum;
    }
}
